docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -168,4 +176,28 @@ test('My test', async t => {
         .useRole(role);
 ```
 
-**Default value**: `false`
\ No newline at end of file
+**Default value**: `false`
+
+## Troubleshooting
+
+### Test Actions Fail After Authentication
+
+This issue can be caused by the browser's page caching. Try disabling it and check if the issue persists.
+
+Use the [fixture.disablePageCaching](../test-code-structure.md#disable-page-caching) and [test.disablePageCaching](../test-code-structure.md#disable-page-caching) methods to disable caching during a particular fixture or test.
+
+To disable page caching during the entire test run, use either of the following options:
+
+* the [--disable-page-caching](../../using-testcafe/command-line-interface.md#--disable-page-caching) command line flag
+* the `disablePageCaching` option in the [runner.run](../../using-testcafe/programming-interface/runner.md#run) method
+* the [disablePageCaching](../../using-testcafe/configuration-file.md#disablepagecaching) configuration file property
+
+#### Why This Issue Occurs
+
+In order to keep tests isolated, TestCafe prevents the tested page from accessing actual browser's local and session storages or cookies. Instead, it redirects all the respective API calls to an internal object (a *sandbox*) injected into the page code. This sandbox tracks any modifications to the cookies and storages, and keeps their up-to-date versions.
```

Comment:
See my variant:

TestCafe starts role initialization with the empty cookie, local storage and session storages. During login actions, the page state (cookie and browser storages) has changed. If redirect occurs, TestCafe stores the current page state on the server and then restore this state after redirect (to guarantee valid state of the client-side of the browser). So, to work properly TestCafe requires that all request goes through the server. The login process of real-world applications can be quite complicated and includes several page redirects. If during the login process some page is cached it can break the normal functioning of the application. The 'disable-page-caching' feature can make Testcafe work correctly in the case then there are caching pages during the login process.